platforms/documentation/docs/src/docs/userguide/dep-man/dependency-management/dependency_locking.adoc

```diff
@@ -0,0 +1,301 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[dependency-locking]]
+= Locking Versions
+
+Using <<dynamic_versions.adoc#sub:declaring_dependency_with_dynamic_version, dynamic dependency versions>> (e.g., `1.+` or `[1.0,2.0)`) can make builds non-deterministic, causing them to break unexpectedly, sometimes due to a transitive dependency outside the build author's control.
+
+====
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-dynamic/kotlin",files="build.gradle.kts[tags=dynamic]"]
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-dynamic/groovy",files="build.gradle[tags=dynamic]"]
+====
+
+To ensure https://reproducible-builds.org/[reproducible builds], it's necessary to *lock versions* of dependencies and their transitive dependencies.
+This guarantees that a build with the same inputs will always resolve to the same module versions, a process known as *dependency locking*.
+
+Dependency locking is crucial for several scenarios:
+
+- **Managing Multiple Repositories:** Companies managing multiple repositories can avoid relying on `-SNAPSHOT` or changing dependencies, which can cause cascading failures when a dependency introduces a bug or incompatibility.
+
+- **Using the Latest Dependencies:** Teams that use dynamic versions can lock their dependencies only for releases. The release tag will include the locked states, ensuring the build remains fully reproducible when bug fixes need to be developed.
+
+- **Publishing Resolved Versions:** Combined with <<publishing_maven.adoc#publishing_maven:resolved_dependencies,publishing resolved versions>>, dependency locking allows replacing the declared dynamic versions at publication time. Consumers will then see the resolved versions that your release used.
+
+== Locking specific configurations
+
+Locking is enabled per <<dependency_configurations.adoc#sec:what-are-dependency-configurations,dependency configuration>>.
+
+Once enabled, you must create an initial lock state, causing Gradle to verify that resolution results do not change.
```

Comment:
💅 These next few sections just cover _activating locking_, not creating the initial lock state.  Maybe the titles could be explicit about that?  Like:

"Locking in Gradle is done on a per-Configuration basis.  The first step towards using locking is to activate locking on whatever configurations..."

Reply:
I update the section header to reflect this.